<robot xmlns:xacro="http://ros.org/wiki/xacro"
        xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#slider"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
       xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body"
       xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">

  <xacro:include filename="$(find sr_description)/hand/xacro/thumb/thmiddle.gazebo.xacro" />
  <xacro:include filename="$(find sr_description)/hand/xacro/thumb/thmiddle.transmission.xacro" />
  
  <!-- muscle is true for muscle hand or false for motor hand -->
  <!-- bio is true for biotac finger or false for standard finger -->
  <!-- prefix can be anything but usually is "r_" or  "l_" for right and left hands distinction -->
  <!-- parent is the parent link for the first joint -->
  <xacro:macro name="thmiddle" params="muscle bio prefix parent">
    <link name="${prefix}thmiddle">
      <inertial>
        <mass value="0.027" />
        <origin xyz="0 0 0.016" rpy="0 0 0"/>
          <inertia  ixx="0.0026" ixy="0.0"  ixz="0.0"  iyy="0.0026"  iyz="0.0" izz="0.000169" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <xacro:if value="${bio}">
        <geometry name="${prefix}thmiddle_visual">
          <mesh filename="package://sr_description/hand/model/biotacs/biotac_thumb_adapter.dae" scale="0.001 0.001 0.001" />
        </geometry>
        <material name="BiotacGreen" />
        </xacro:if>
        <xacro:unless value="${bio}">
        <geometry name="${prefix}thmiddle_visual">
          <mesh filename="package://sr_description/hand/model/TH2_z.dae" scale="0.001 0.001 0.001" />
        </geometry>
        <material name="shadow_thmiddle_material">
          <color rgba="0.2 0.2 0.2 1.0"/>
        </material>
        </xacro:unless> 
      </visual>  
      <collision>
        <origin xyz="0 0 0.012" rpy="0 0 0 " />
        <geometry name="${prefix}thmiddle_collision_geom">
          <xacro:if value="${bio}">
            <box size="0.008 0.008 0.01" />
          </xacro:if>
          <xacro:unless value="${bio}">
            <cylinder radius="0.011" length="0.018" />
          </xacro:unless> 
        </geometry>
      </collision>
      <xacro:unless value="${bio}">
     <collision>
        <origin xyz="0 0 0.019" rpy="0 0 0 " />
        <geometry>
          <sphere radius="0.011" />
        </geometry>
      </collision> 
      </xacro:unless> 
      
    </link>
    
    <joint name="${prefix}THJ2" type="revolute">
      <parent link="${parent}"/>
      <child link="${prefix}thmiddle"/>
				<xacro:if value="${bio}">
					<origin xyz="0 0 0.01" rpy="0 0 0" /> <!-- FIXME: z offset is too large -->
				</xacro:if>
				<xacro:unless value="${bio}">
					<origin xyz="0 0 0" rpy="0 0 0" />
				</xacro:unless> 
      <axis xyz="0 -1 0" />
      <limit lower="${-40/180*M_PI}" upper="${40/180*M_PI}" effort="10"
      velocity="1.0" />
      <dynamics damping="0.001"/> 
    </joint>

		<!-- extensions -->
		<xacro:thmiddle_transmission muscletrans="${muscle}" prefix="${prefix}"/>
		<xacro:thmiddle_gazebo prefix="${prefix}" />
  </xacro:macro>

</robot>
